File: results-05-12/python/python-00001-benchmark-oauth/qwen2-5-coder-32b/editor/final-src-code.py

import os
import json
os.environ["SHINKAI_NODE_LOCATION"] = "http://localhost:9950"
os.environ["BEARER"] = "debug"
os.environ["X_SHINKAI_TOOL_ID"] = "tool-id-debug"
os.environ["X_SHINKAI_APP_ID"] = "tool-app-debug"
os.environ["X_SHINKAI_LLM_PROVIDER"] = "o_qwen2_5_coder_32b"
os.environ["HOME"] = "results/python/python-00001-benchmark-oauth/qwen2-5-coder-32b/editor/home"
os.environ["MOUNT"] = "results/python/python-00001-benchmark-oauth/qwen2-5-coder-32b/editor/mount"
os.environ["ASSETS"] = "results/python/python-00001-benchmark-oauth/qwen2-5-coder-32b/editor/assets"

from typing import Dict, Any, Optional, List
from google.oauth2.credentials import Credentials
from google_auth_oauthlib.flow import InstalledAppFlow
from googleapiclient.discovery import build
import os

class CONFIG:
    client_secret_file: str
    token_file: str
    scopes: List[str]

class INPUTS:
    event: str
    description: str
    start_iso: str
    end_iso: str

class OUTPUT:
    event_id: Optional[str]
    error_message: Optional[str]

async def run(config: CONFIG, inputs: INPUTS) -> OUTPUT:
    creds = None
    if os.path.exists(config.token_file):
        creds = Credentials.from_authorized_user_file(config.token_file, config.scopes)
    
    if not creds or not creds.valid:
        if creds and creds.expired and creds.refresh_token:
            creds.refresh(Request())
        else:
            flow = InstalledAppFlow.from_client_secrets_file(
                config.client_secret_file, config.scopes)
            creds = flow.run_local_server(port=0)
        
        with open(config.token_file, 'w') as token:
            token.write(creds.to_json())

    try:
        service = build('calendar', 'v3', credentials=creds)

        event = {
            'summary': inputs.event,
            'description': inputs.description,
            'start': {
                'dateTime': inputs.start_iso,
                'timeZone': 'UTC',
            },
            'end': {
                'dateTime': inputs.end_iso,
                'timeZone': 'UTC',
            },
        }

        event = service.events().insert(calendarId='primary', body=event).execute()
        return OUTPUT(event_id=event['id'], error_message=None)
    except Exception as e:
        return OUTPUT(event_id=None, error_message=str(e))

if __name__ == "__main__":
    import asyncio
    
    config = CONFIG()
    inputs = INPUTS()
    inputs.event="Test Event"
    inputs.description="Test Description"
    inputs.start_iso="2024-12-05T10:00:00"
    inputs.end_iso="2024-12-05T11:00:00"
    
    # Run the async function
    result = asyncio.run(run(config, inputs))
    print(json.dumps(result.__dict__))

File: results-05-12/typescript/typescript-00012-benchmark-oauth/qwen2-5-coder-32b/editor/final-src-code.ts

  // These environment variables are required, before any import.
  // Do not remove them, as they set environment variables for the Shinkai Tools.
  Deno.env.set('SHINKAI_NODE_LOCATION', "http://localhost:9950");
  Deno.env.set('BEARER', "debug");
  Deno.env.set('X_SHINKAI_TOOL_ID', "tool-id-debug");
  Deno.env.set('X_SHINKAI_APP_ID', "tool-app-debug");
  Deno.env.set('X_SHINKAI_LLM_PROVIDER', "o_qwen2_5_coder_32b");
  Deno.env.set('HOME', "results/typescript/typescript-00012-benchmark-oauth/qwen2-5-coder-32b/editor/home");
  Deno.env.set('MOUNT', "results/typescript/typescript-00012-benchmark-oauth/qwen2-5-coder-32b/editor/mount");
  Deno.env.set('ASSETS', "results/typescript/typescript-00012-benchmark-oauth/qwen2-5-coder-32b/editor/assets");
  
import { getHomePath } from './shinkai-local-support.ts';

type CONFIG = {
  client_id: string;
  client_secret: string;
  redirect_uri: string;
  refresh_token: string;
};

type INPUTS = {
  event: string;
  description: string;
  start_iso: string;
  end_iso: string;
};

type OUTPUT = {
  success: boolean;
  message: string;
};

const TOKEN_URL = 'https://oauth2.googleapis.com/token';
const CALENDAR_API_URL = 'https://www.googleapis.com/calendar/v3/calendars/primary/events';

async function getAccessToken(config: CONFIG): Promise<string> {
  const response = await fetch(TOKEN_URL, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
    },
    body: new URLSearchParams({
      client_id: config.client_id,
      client_secret: config.client_secret,
      refresh_token: config.refresh_token,
      grant_type: 'refresh_token',
    }),
  });

  if (!response.ok) {
    throw new Error(`Failed to fetch access token: ${await response.text()}`);
  }

  const data = await response.json();
  return data.access_token;
}

async function createEvent(config: CONFIG, inputs: INPUTS): Promise<OUTPUT> {
  try {
    const accessToken = await getAccessToken(config);
    const event = {
      summary: inputs.event,
      description: inputs.description,
      start: { dateTime: inputs.start_iso },
      end: { dateTime: inputs.end_iso },
    };

    const response = await fetch(CALENDAR_API_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${accessToken}`,
      },
      body: JSON.stringify(event),
    });

    if (!response.ok) {
      throw new Error(`Failed to create event: ${await response.text()}`);
    }

    return { success: true, message: 'Event created successfully' };
  } catch (error: unknown) {
    return { success: false, message: String(error) };
  }
}

export async function run(config: CONFIG, inputs: INPUTS): Promise<OUTPUT> {
  return createEvent(config, inputs);
}

  
  // console.log('Running...')
  // console.log('Config: {}')
  // console.log('Inputs: {"event":"Test Event","description":"Test Description","start_iso":"2024-12-05T10:00:00","end_iso":"2024-12-05T11:00:00"}')
  
  try {
    const program_result = await run({}, {"event":"Test Event","description":"Test Description","start_iso":"2024-12-05T10:00:00","end_iso":"2024-12-05T11:00:00"});
    if (program_result) console.log(JSON.stringify(program_result, null, 2));
    else console.log(program_result);
  } catch (e) {
    console.log('::ERROR::', e);
  }

